wrappers/src/dictionary_cs.cpp

```diff
@@ -71,6 +71,33 @@ extern "C" {
         });
     }
 
+    REALM_EXPORT void* realm_dictionary_add_collection(object_store::Dictionary& dictionary, realm_value_t key, realm_value_type type, NativeException::Marshallable& ex)
```

Comment:
We could do it, but I think it's going to be a little bit less readable. Either we will have an if inside each branch of the switch, or we use something like a function pointer that we set once before the switch

Reply:
Why do we need an if inside each branch? Maybe I'm overlooking something since they're so similar. But the way I was thinking about it would be something like:

```cpp
REALM_EXPORT void* realm_dictionary_add_collection(object_store::Dictionary& dictionary, realm_value_t key, realm_value_type type, bool allow_value_override, NativeException::Marshallable& ex) {
    return handle_errors(ex, [&]()-> void* {
        auto dict_key = from_capi(key.string);
        if (!allow_value_override && dictionary.contains(dict_key))
        {
            throw KeyAlreadyExistsException(dict_key);
        }

        switch (type) {
            ...
        }
    });
}
```

Reply:
Oh yes, you're right. This was comment was supposed to go on the similar comment about lists. For lists, `set` and `insert` use two different methods inside, differently from the dictionary here. Sorry for the confusion.

Reply:
Ugh, yeah, for lists it's a bit more annoying. We could still merge them and pass the method as an argument, but also we can probably leave them as they are. Don't have strong opinions on the matter.